libs/k8s-client/src/KubernetesApiClientFacade.php

```diff
@@ -227,29 +261,68 @@ public function deleteAllMatching(?DeleteOptions $deleteOptions = null, array $q
         }
     }
 
+    /**
+     * @template T of ConfigMap|Event|PersistentVolumeClaim|Pod|Secret|Service|Ingress|PersistentVolume
+     * @param class-string<T> $resourceType
+     */
+    public function checkResourceExists(string $resourceType, string $resourceName): bool
+    {
+        try {
+            $this->getApiForResource($resourceType)->get($resourceName);
+            return true;
+        } catch (ResourceNotFoundException $e) {
+        }
+
+        return false;
+    }
+
     /**
      * @param class-string<AbstractModel> $resourceType
      * @return ($resourceType is class-string<ConfigMap> ? ConfigMapsApiClient :
      *         ($resourceType is class-string<Event> ? EventsApiClient :
-     *         ($resourceType is class-string<PersistentVolumeClaim> ? PersistentVolumeClaimApiClient :
+     *         ($resourceType is class-string<PersistentVolumeClaim> ? PersistentVolumeClaimsApiClient :
      *         ($resourceType is class-string<Pod> ? PodsApiClient :
      *         ($resourceType is class-string<Secret> ? SecretsApiClient :
-     *         never))))))
+     *         ($resourceType is class-string<Service> ? ServicesApiClient :
+     *         ($resourceType is class-string<Ingress> ? IngressesApiClient :
+     *         ($resourceType is class-string<PersistentVolume> ? PersistentVolumesApiClient :
+     *         never))))))))
      */
     // phpcs:ignore Generic.Files.LineLength.MaxExceeded
-    private function getApiForResource(string $resourceType): ConfigMapsApiClient|EventsApiClient|PersistentVolumeClaimApiClient|PodsApiClient|SecretsApiClient
+    private function getApiForResource(string $resourceType): ConfigMapsApiClient|EventsApiClient|PersistentVolumeClaimsApiClient|PodsApiClient|SecretsApiClient|ServicesApiClient|IngressesApiClient|PersistentVolumesApiClient
+    {
+        $resourceMap = $this->getResrouceTypeClientMap();
```

Comment:
Hele, tohle se mi uplne nezda. Chapu, ze si se chtel zbavit toho mit 2x seznam resource types, ale `getApiForResource` se vola docela casto (obcas i v nejakem foreach) a pro kazde zavolani se ted bude vytvaret a zahazovat to pole, jen aby se z nej vybral jeden item. Navrhoval by z toho teda aspon udela property, zinicializovat ji jednou v konstruktoru?

Reply:
:) opraveno